spec/features/publish/new_course_spec.rb

```diff
@@ -0,0 +1,243 @@
+
+    save_course
+  end
+
+  def given_i_am_authenticated_as_a_provider_user
+    given_i_am_authenticated(
+      user: create(
+        :user,
+        providers: [
+          create(:provider, :accredited_body, sites: [build(:site), build(:site)]),
+        ],
+      ),
+    )
+  end
+
+  def when_i_visit_the_courses_page
+    publish_provider_courses_index_page.load(
+      provider_code: provider.provider_code, recruitment_cycle_year: provider.recruitment_cycle_year,
+    )
+  end
+
+  def provider
+    @provider ||= @current_user.providers.first
+  end
+
+  def course
+    @course ||= provider.courses.first
+  end
+
+  def recruitment_cycle
+    @recruitment_cycle ||= provider.recruitment_cycle
+  end
+
+  def sites
+    @sites ||= provider.sites.sort_by(&:location_name)
+  end
+
+  def and_i_click_on_add_course
+    publish_provider_courses_index_page.add_course.click
+  end
+
+  def then_i_see_the_new_course_level_page
+    expect(page).to have_current_path("/publish/organisations/#{provider.provider_code}/#{provider.recruitment_cycle_year}/courses/level/new")
+  end
+
+  def expect_page_to_be_displayed_with_query(page:, expected_query_params:)
+    current_query_string = current_url.match('\?(.*)$').captures.first
+    url_params = { course: expected_query_params }
+
+    expect(page).to be_displayed
+
+    query = Rack::Utils.parse_nested_query(current_query_string).deep_symbolize_keys
+
+    expect(query).to match(url_params)
+  end
+
+  def select_level(course_creation_params, level:, level_selection:, next_page:)
+    course_creation_params[:level] = level
+    course_creation_params[:is_send] = "0"
+
+    level_selection.click
+    new_level_page.continue.click
+
+    expect_page_to_be_displayed_with_query(
+      page: next_page,
+      expected_query_params: course_creation_params,
+    )
+
+    course_creation_params
+  end
+
+  def primary
+    @primary ||= %i[primary primary_with_english primary_with_geography_and_history primary_with_mathematics primary_with_modern_languages primary_with_physical_education primary_with_science].sample
+  end
+
+  def primary_subject
+    @primary_subject ||= find(:primary_subject, primary)
+  end
+
+  def course_subject(level)
+    if level == "primary"
+      primary_subject
+    end
+  end
+
+  def select_subjects(course_creation_params, level:, next_page:)
+    course_creation_params[:level] = level
+    course_subject = course_subject(level)
+    course_creation_params[:subjects_ids] = [course_subject.id.to_s]
+
+    new_subjects_page.subjects_fields.select(course_subject.subject_name)
+    new_subjects_page.continue.click
+
+    expect_page_to_be_displayed_with_query(
+      page: next_page,
+      expected_query_params: course_creation_params,
+    )
+
+    course_creation_params
+  end
+
+  def select_age_range(course_creation_params, next_page:)
+    course_creation_params[:age_range_in_years] = "5_to_11"
+
+    choose("course_age_range_in_years_5_to_11")
```

Comment:
Does this not have a page object?

Reply:
done in https://github.com/DFE-Digital/teacher-training-api/pull/2525/commits/77827debc4ed2be38058f60183926ff5264ee34a